src/reflect/scala/reflect/internal/TreeGen.scala

```diff
@@ -143,11 +143,15 @@ abstract class TreeGen {
 
   /** Builds a reference to given symbol with given stable prefix. */
   def mkAttributedRef(pre: Type, sym: Symbol): RefTree = {
-    val qual = mkAttributedQualifier(pre)
-    qual match {
-      case EmptyTree                                  => mkAttributedIdent(sym)
-      case This(clazz) if qual.symbol.isEffectiveRoot => mkAttributedIdent(sym)
-      case _                                          => mkAttributedSelect(qual, sym)
+    pre match {
+      case TypeRef(NoPrefix, _: TypeSkolem, Nil) => mkAttributedIdent(sym)
```

Comment:
I apologize for not leaving a comment here earlier. I was thinking about a nicer way of doing this, but rather than wait until I have a concrete suggestion, I'll already say this special case makes me uneasy. Why does a skolem show up? Should the type have been deskolemized before we pass it to `mkAttributedRef`?